service/matching/forwarder.go

```diff
@@ -266,6 +263,9 @@ func (fwdr *Forwarder) ForwardPoll(ctx context.Context) (*internalTask, error) {
 // that's necessary before making a ForwardTask or ForwardQueryTask API call.
 // After the API call is invoked, token.release() must be invoked
 func (fwdr *Forwarder) AddReqTokenC() <-chan *ForwarderReqToken {
+	if fwdr == nil {
+		return nil
+	}
```

Comment:
Are we seeing cases of invoking these member functions on a `nil` pointer?

Reply:
It's a little simplification to get rid of the separate fwdr{Poll,Add}ReqTokenC functions. See the individual commit https://github.com/temporalio/temporal/pull/1708/commits/1af3a4fbaffe5c38dcf565beb6f35cec0b576dfe